### ISSUE TITLE
Problems defining variables with `const/let` in `:on` attributes

### ISSUE THREAD

**Issue by thebug404** [dy/sprae#5]
## Describe problem. 🚨

When you try to set events to HTML elements, many times we find ourselves in the need to define variables to store certain information. Here is an example.

```html
<main>
  <button :onclick="() => {
    const username = 'thebug'
    console.log(username)
  }">Click me!</button>
</main>
```

The problem that happens when doing is that our function will never be executed. And this happens when we use `const/let`

## Why it happens? 🤔

**Sprea** has a function that is responsible for parsing expressions. That function is called `parseExpr`.

What happens with this function is that it evaluates when a directive is prefixed with `:on`, and what it does is wrap our function in a self-executing function.

Below is the source code. (Take a look at line 426)

https://github.com/dy/sprae/blob/82e2b20df496eb3a959bf4db1d08fae091fc0338/src/directives.js#L421-L427

The problem with these is that we can't use `const` or `let` to declare variables in the expression evaluated in the `parseExpr` function. This is because the expression is compiled into an anonymous function that is executed at runtime and does not have a block of code in which a block-scoped variable such as `const` or `let` can be declared.

## Solution. 💡

We have several alternatives to solve this problem, they are shown below.

**First option**

Declare our values in a global variable.

```html
<main>
  <button :onclick="() => {
    window.username = 'thebug'
    console.log(window.username)
  }">Click me!</button>
</main>
```

> Note: that this can cause a conflict when setting new properties to the global object.

**Second option**

It is to store the values by referencing the element.

```html
<main>
  <button :onclick="() => {
    this.username = 'thebug'
    console.log(this.username)
  }">Click me!</button>
</main>
```

> Note: that this may conflict with the element's reserved attributes.

**Third option**

Declare variables using `var`

```html
<main>
  <button :onclick="() => {
    var username = 'thebug'
    console.log(username)
  }">Click me!</button>
</main>
```

## Conclusion. 💪🏻

In conclusion, the `parseExpr` function performs a check to avoid errors during the evaluation of the expression, and therefore you should declare the variable in the global scope using the `window`, `this`, or `var` object instead of using `const/let`. It is important to consider the scope implications of each type of variable declaration to avoid errors and ensure that variables are available where they are needed.

**Comment by dy** [comment]
Oh, thanks for the catch! Will fix promptly

**Comment by dy** [comment]
Fixed in 2.14.2.

**Comment by thebug404** [comment]
Thanks bro, you're amazing. 👌
